### PR TITLE
Add dependency check to CI

### PR DESCRIPTION
**Pull request by troydai**
Add a task to the Travis CI which install products and list the external dependencies to compare to the recorded dependency list in the source repository. The goal of the task is to notify the change of the external dependencies which are explicitly or implicitly changed.

As of now the change of the dependencies will not fail the CI tasks. That will come in the future when an automatic rebase mechanism is introduced. 

Additionally, the verify dependency step is moved from the integration check step to here as well.

### CODE REVIEW

File: ext.txt [hunk 1]
```diff
@@ -0,0 +1,101 @@
+adal==1.0.2
```
Comment (tjprescott) [inline ext.txt]: Why is this here twice?

File: scripts/dependency/check.sh [hunk 2]
```diff
@@ -0,0 +1,44 @@
+#!/bin/bash
+
+set -e
```
Comment (tjprescott) [inline scripts/dependency/check.sh]: This script should be part of `azdev verify` because this script will not run on Windows.

Comment (tjprescott) [reply scripts/dependency/check.sh]: Also, like we discussed in scrum, this should execute along with Derek's dependency checks.

File: ext.txt (hunk 1 above)
Comment (troydai) [reply ext.txt]: will be deleted.

**Comment by troydai** [comment]
@tjprescott please take another look.

1. I will not roll this script into `azdev` because I don't want to install `azdev` and pull in unexpected dependency.
2. This change will not fail CI at this point.
